cmd/soroban-cli/src/snapshot.rs

```diff
@@ -120,10 +127,85 @@ where
         protocol_version: ledger_info.protocol_version,
         sequence_number: ledger_info.sequence_number,
         timestamp: ledger_info.timestamp,
-        network_passphrase: ledger_info.network_passphrase,
+        network_passphrase: ledger_info.network_passphrase.clone(),
         base_reserve: ledger_info.base_reserve,
     };
     serde_json::to_writer(&file, &output)?;
 
     Ok(())
 }
+
+/// Returns a list of events from the on-disk event store, which stores events
+/// exactly as they'd be returned by an RPC server.
+pub fn read_events(path: &std::path::PathBuf) -> Result<Vec<rpc::Event>, Error> {
+    let reader = std::fs::OpenOptions::new().read(true).open(path)?;
+    let events: rpc::GetEventsResponse = serde_json::from_reader(reader)?;
+
+    Ok(events.events)
+}
+
+// Reads the existing event file, appends the new events, and writes it all to
+// disk. Note that this almost certainly isn't safe to call in parallel.
+pub fn commit_events(
+    new_events: &[events::HostEvent],
+    ledger_info: &LedgerInfo,
+    output_file: &std::path::PathBuf,
+) -> Result<(), Error> {
+    // Create the directory tree if necessary, since these are unlikely to be
+    // the first events.
+    if let Some(dir) = output_file.parent() {
+        if !dir.exists() {
+            create_dir_all(dir)?;
+        }
+    }
+
+    let mut file = std::fs::OpenOptions::new().read(true).open(output_file)?;
+    let mut events: rpc::GetEventsResponse = serde_json::from_reader(&mut file)?;
+
+    for event in new_events.iter() {
+        let contract_event = match event {
+            events::HostEvent::Contract(e) => e,
+            events::HostEvent::Debug(_e) => todo!(),
+        };
+
+        // TODO: Handle decoding errors cleanly; I miss errors.Wrap(err, ...) :(
+        let topics = match &contract_event.body {
+            xdr::ContractEventBody::V0(e) => &e.topics,
+        }
+        .iter()
+        .map(|t| t.to_xdr_base64().unwrap())
+        .collect(); // try_collect()? would be nice here
+
+        let cereal_event = rpc::Event {
+            event_type: "contract".to_string(),
+            id: String::new(),
+            paging_token: String::new(),
```

Comment:
Should be an id, no?

Reply:
Yep, good catch: added in fe2a78a alongside a toid package.